src/clj/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_core.clj

```diff
@@ -126,9 +146,9 @@
     (let [stopped? (atom false)]
       (shutdown-fn #(while (not @stopped?)
                      (try
-                       (let [[watch-key events] (retrieve-events watcher)]
-                         (when-not (empty? events)
-                           (process-events! watcher watch-key events)))
+                       (let [events-by-key (retrieve-events watcher)]
+                         (when-not (empty? (vals events-by-key))
```

Comment:
What will `events-by-key` be when it has no vals? Could you have a key in events-by-key without any values?

Reading the `retrieve-events` function, I'm not seeing a way for it to return with an empty return, as it blocks on the take until there are events to report.